### PR TITLE
feat: new lacework_query resource

### PR DESCRIPTION
**Pull request by dmurray-lacework**
***Issue***: https://lacework.atlassian.net/browse/ALLY-870

***Description:***
Add new resource `lacework_query`

Usage:
```hcl 
 resource "lacework_query" "example" {
    query_id       = var.query_id
    evaluator_id   = "Cloudtrail"
    query          = var.query
  }
  ```

### CODE REVIEW

File: examples/resource_lacework_query/main.tf [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    lacework = {
+      source = "lacework/lacework"
+    }
+  }
+}
+
+  resource "lacework_query" "example" {
+    id    = var.query_id
+    query = <<EOT
+    {
```
Comment (afiune) [inline examples/resource_lacework_query/main.tf:12]: This type of query won't work until we release the feature request to have anonymous queries.

File: website/docs/r/query.html.markdown [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+The following arguments are supported:
+
+* `id` - (Required) The query id.
+* `query` - (Required) The query string.
+* `evauator_id` - (Optional) The query evaluator id.
+
+## Import
+
+A Lacework query can be imported using a `QUERY_ID`, e.g.
```
Comment (afiune) [inline website/docs/r/query.html.markdown:49]: Please, add the CLI commands to list queries:
```
lacework query list
```

File: examples/resource_lacework_query/main.tf [hunk 3]
```diff
@@ -0,0 +1,40 @@
+terraform {
+  required_providers {
+    lacework = {
+      source = "lacework/lacework"
+    }
+  }
+}
+
+  resource "lacework_query" "example" {
```
Comment (hazedav) [inline examples/resource_lacework_query/main.tf:9]: nit. Is there a leading space here?

File: examples/resource_lacework_query/main.tf [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+    query_id       = var.query_id
+    evaluator_id   = "Cloudtrail"
+    query          = var.query
```
Comment (hazedav) [inline examples/resource_lacework_query/main.tf:12]: nit.  should we call this query_text to be consistent with Lacework APIs and SDK structs?

**Review by hazedav — APPROVED** [review]
Approved w/ comments.

File: examples/resource_lacework_query/main.tf (hunk 3 above)
Comment (afiune) [reply examples/resource_lacework_query/main.tf:9]: yes, let's format this file, please.

**Review by afiune — CHANGES_REQUESTED** [review]
Unfortunately, when we do not specify an evaluator ID, the platform  returns the string `"<<IMPLICIT>>"` which causes a constant drift. 

Screen Shot 2022-02-16 at 11 50 57 PM

How do we address this @dmurray-lacework @hazedav ?

**Comment by dmurray-lacework** [comment]
> Unfortunately, when we do not specify an evaluator ID, the platform returns the string `"<<IMPLICIT>>"` which causes a constant drift.
> 
> Screen Shot 2022-02-16 at 11 50 57 PM
> 
> How do we address this @dmurray-lacework @hazedav ?

We could set a default value of "<<IMPLCIT>>" in the schema. 
Or a diffsupressfunc

I'll take a look at this today

File: lacework/resource_lacework_query.go [hunk 5]
```diff
@@ -0,0 +1,163 @@
...
+		QueryText: d.Get("query").(string),
+	}
+
+	log.Printf("[INFO] Updating Query with data:\n%+v\n", query)
+	response, err := lacework.V2.Query.Update(d.Id(), query)
+	if err != nil {
+		return err
+	}
+
+	d.SetId(response.Data.QueryID)
```
Comment (afiune) [inline lacework/resource_lacework_query.go:126]: I don't think ID's can be changed, lets's not "update" it
```suggestion
```

File: lacework/resource_lacework_query.go [hunk 6]
```diff
@@ -0,0 +1,163 @@
...
+		lacework = meta.(*api.Client)
+	)
+
+	log.Printf("[INFO] Reading Query with guid %s\n", d.Id())
+	response, err := lacework.V2.Query.Get(d.Id())
+	if err != nil {
+		return err
+	}
+
+	d.SetId(response.Data.QueryID)
```
Comment (afiune) [inline lacework/resource_lacework_query.go:99]: I also don't think that Read should set the ID. I just went and looked and other resources in our codebase have this pattern but it is wrong.
```suggestion
```

File: website/docs/r/query.html.markdown [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+LQL is a human-readable text syntax for specifying selection, filtering, and manipulation of data. 
+It overlaps with SQL in its syntax and what it allows.
+
+For more information, see the [LQL Overview Documentation](https://docs.lacework.com/lql-overview).
+
+## Example Usage
+
+```hcl
+  resource "lacework_query" "example" {
+  query_id       = Lql_Terraform_Query
```
Comment (afiune) [inline website/docs/r/query.html.markdown:21]:
```suggestion
  query_id       = "Lql_Terraform_Query"
```

File: website/docs/r/query.html.markdown [hunk 8]
```diff
@@ -0,0 +1,64 @@
...
+
+# lacework\_query
+
+To provide customizable specification of datasources, Lacework provides the Lacework Query Language (LQL). 
+LQL is a human-readable text syntax for specifying selection, filtering, and manipulation of data. 
+It overlaps with SQL in its syntax and what it allows.
+
+For more information, see the [LQL Overview Documentation](https://docs.lacework.com/lql-overview).
+
+## Example Usage
```
Comment (afiune) [inline website/docs/r/query.html.markdown:17]: I wish we could add at least two examples, one that doesn't have evaluator ID and another one that NEEDS this evaluator ID and explain why the user needs it.

**Review by afiune — CHANGES_REQUESTED** [review]

File: lacework/resource_lacework_query.go [hunk 9]
```diff
@@ -0,0 +1,166 @@
...
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "The query string",
+			},
+			"evaluator_id": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "The query evaluator id",
+				DiffSuppressFunc: func(k, old, new string, d *schema.ResourceData) bool {
+					return old == "<<IMPLICIT>>" || new == "<<IMPLICIT>>"
```
Comment (afiune) [inline lacework/resource_lacework_query.go:39]: I'm not sure that after this we could change the evaluator ID but I also think those are edge cases where the user wants to repurpose a query, so let us ignore this for now. 😅

**Review by afiune — APPROVED** [review]
Two comments but out of that...

tenor-128082851

File: lacework/resource_lacework_query.go [hunk 10]
```diff
@@ -0,0 +1,166 @@
...
+	log.Printf("[INFO] Read Query with guid %s\n", response.Data.QueryID)
+	return nil
+}
+
+func resourceLaceworkQueryUpdate(d *schema.ResourceData, meta interface{}) error {
+	var (
+		lacework = meta.(*api.Client)
+	)
+
+	if d.HasChange("query_id") {
```
[the remark this answers is not in the document]
Comment (dmurray-lacework) [reply lacework/resource_lacework_query.go:116]: I can't find a way to check the diff between old and new on validation.

**Pull request merged**